cpp/src/arrow/acero/query_context.cc

```diff
@@ -23,6 +23,36 @@ namespace arrow {
 using arrow::internal::CpuInfo;
 namespace acero {
 
+namespace internal {
+
+int64_t GetTempStackSizeFromEnvVar() {
+  auto maybe_env_value = arrow::internal::GetEnvVar(kTempStackSizeEnvVar);
+  if (!maybe_env_value.ok()) {
+    return kDefaultTempStackSize;
+  }
+  auto env_value = *std::move(maybe_env_value);
```

Comment:
Being explicit about the type here would be helpful:

```cpp
std::string env_value = std::move(maybe_env_value).ValueUnsafe();
```

Reply:
Thanks. Addressed.